hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/clustering/ClusteringPlanOperator.java

```diff
@@ -88,10 +96,20 @@ public void notifyCheckpointComplete(long checkpointId) {
   }
 
   private void scheduleClustering(HoodieFlinkTable<?> table, long checkpointId) {
+    List<HoodieInstant> pendingClusteringInstantTimes =
+        ClusteringUtils.getPendingClusteringInstantTimes(table.getMetaClient());
     // the first instant takes the highest priority.
     Option<HoodieInstant> firstRequested = Option.fromJavaOptional(
-        ClusteringUtils.getPendingClusteringInstantTimes(table.getMetaClient()).stream()
+        pendingClusteringInstantTimes.stream()
             .filter(instant -> instant.getState() == HoodieInstant.State.REQUESTED).findFirst());
+
+    long pendingClusteringCount = pendingClusteringInstantTimes.stream()
```

Comment:
Why do we only include inflight instants? Will inflight and requested instants be double-counted? If that's the case, we should also correct the method `org.apache.hudi.metrics.FlinkCompactionMetrics#setPendingCompactionCount`.

Reply:
@stream2000 hello , I have changed, don't need to modify the org.apache.hudi.metrics.FlinkCompactionMetrics#setPendingCompactionCount